src/ui_kit/src/folder/mod.rs

```diff
@@ -90,16 +73,15 @@ pub fn Folder(cx: Scope<Props>) -> Element {
                             }
                             let drag_file_event_in_app = get_drag_file_event_in_app();
                             if let Some(file_name) = drag_file_event_in_app.file_name {
-                                let root_directory = file_storage.root_directory();  
+                                let current_directory = file_storage.current_directory().unwrap_or_default();  
                                 let folder_name = folder_name_complete_ref.with(|name| name.clone());
-                                let current_directory = root_directory.get_item(&folder_name).unwrap().get_directory().unwrap();
-                                let file = root_directory.get_item(&file_name).unwrap();
-                                match current_directory.add_item(file.clone()) {
+                                let directory_target = current_directory.get_item(&folder_name).unwrap().get_directory().unwrap_or_default();
```

Comment:
might want to do something like 

```suggestion
                                let directory_target = match current_directory.get_item(&folder_name).and_then(|item| item.get_directory()) {
                                      Ok(dir) => dir,
                                      _ => return
                                };
```

Defaulting on the directory would yield a empty directory that would not be apart of the index if it was not able to unwrap which would invalidate the whole thing. In the future it should have like a popup in the event of it happening.

Reply:
Much better, thanks Darius